### DISCUSSION TITLE
Understand the filter-string

### DISCUSSION THREAD

**Discussion by DamienCassou** [minad/consult#922] [Q&A] [answered]
When I use `consult-grep` or one of its sibling commands, the user input is parsed as 2 search strings: `regexps` (passed as argument to grep) and `filter-string`. The user input looks like this: `#regexps#filter-string`. For example, consult documentation says that `#defun#consult`

> Search for "defun" using grep, filter with the word "consult"

I would like to understand what kind of `filter-string` can be used and which part of the code takes care of using this `filter-string`.

**Comment by minad** [comment]
See also https://github.com/minad/consult?tab=readme-ov-file#asynchronous-search.

**Reply by DamienCassou** [reply]
I read that as well before posting but forgot to mention it. Thank you.

**Comment by oantolin** [comment, marked as answer]
The filter string is matched against the results returned by grep using whatever completion styles you have configured, and this happens in emacs, without rerunning grep. I'd say that 90% of the time  I don't even need a filter string, and when I do use one it is either on a long running search I don't want to rerun (not very common for me, I just don't have that much data), or, more commonly, because I want to use either partial-completion or flex matching against the grep results (two completion styles which are awkward to express directly as regexps).

(N.B. As you can imagine I don't actually use either the partial-completion or flex completion styles, but the corresponding matching styles within orderless.)

**Reply by DamienCassou** [reply]
Thank you very much. I now understand what is happening.
